importers/file_meta.py

```diff
@@ -39,45 +40,111 @@ def _do_import(self, doc, filepath):
+
+    for line in linelist:
       match = re.search('\{.+\[(.+)\]\}(.+)\{.+\[(.+)\]\}', line)
-      caption = None
       if match:
         lang_code = match.group(1)
         caption = match.group(2)
       else:
         lang_code = 'en'
         caption = line
+      if not meta['Title']:
+        meta['Title'] = caption
       meta['Caption'].append({
         'lang': lang_code,
         'text': caption
       })
 
+    # Add code for descriptions
+    descs = meta.get('Description')
+    if descs:
+      for line in descs.split('\n'):
+        match = re.search('\{.+\[(.+)\]\}(.+)\{.+\[(.+)\]\}', line)
+        if match:
+          lang_code = match.group(1)
+          desc = match.group(2)
+        else:
+          lang_code = 'en'
+          desc = line
+        foundit = False
+        for ind, cobj in enumerate(meta['Caption']):
+          if cobj['lang'] == lang_code:
+            meta['Caption'][ind]['description'] = desc
+            foundit = True
+        if not foundit:
+          meta['Caption'].append({
+            'lang': lang_code,
+            'text': 'Untitled',
+            'description': desc
+          })
+    meta['Caption'] = json.dumps(meta['Caption'])  # convert caption metadata to valid json
+    if meta['Title'] == '':
+      meta['Title'] = 'Untitled'
+
+    # Deal with Kmaps (SubjectCode gets turned into an array from semicolons by exiftool but other kmap fields do not)
+    if 'SubjectCode' in meta:
+      meta['SubjectCode'] = json.dumps(meta['SubjectCode'])
+
+    if 'SubjectReference' in meta:
+      meta['SubjectReference'] = json.dumps(meta['SubjectReference'])
+
+    if 'IntellectualGenre' in meta:
+      meta['IntellectualGenre'] = json.dumps(meta['IntellectualGenre'].split(';'))
+
+    # Location: Bridge tool shows just "Sub-location" but Exiftool exports both "Location" and "Sub-location"
+    if 'Sub-location' in meta:
+      meta['Sub-location'] = json.dumps(meta['Sub-location'].split(';'))
+
+    if 'Location' in meta:
+      meta['Location'] = json.dumps(meta['Location'].split(';'))
+    elif 'Sub-location' in meta:
+      meta['Location'] = meta['Sub-location']
+
+    if 'Source' in meta:
+      meta['Source'] = json.dumps(meta['Source'].split(';'))
+
+    if 'ImageCreatorID' in meta:
+      meta['ImageCreatorID'] = meta['ImageCreatorID'].split(';') if isinstance(meta['ImageCreatorID'], str) \
+        else meta['ImageCreatorID']
+      meta['ImageCreatorID'] = json.dumps(meta['ImageCreatorID'])
+
+    if 'ImageCreatorName' in meta:
+      meta['ImageCreatorName'] = meta['ImageCreatorName'].split(';') if isinstance(meta['ImageCreatorName'], str) \
+        else meta['ImageCreatorName']
+      meta['ImageCreatorName'] = json.dumps(meta['ImageCreatorName'])
+
+    # Convert all values in meta dictionary to strings (ndg added mistakenly, leaving temporarily)
+    # for ky in meta.keys():
+    #   meta[ky] = str(meta[ky])
+    with open('../logs/{}.json'.format(meta['Filename']), 'w') as jout:
```

Comment:
Like noted above it might be nice to let logging be optional.

Reply:
Yes, I implemented this as you suggested.

Reply:
So I wasn't clear enough here. I moved this code up to the `_do_import()` method in  `file_meta.py` and I just now added the test for the quiet parameter. So it looks like this:

```
    if self.verbose and not self.quiet:
      jfnm = "{}_{}.json".format(mdata['Filename'].split('.').pop(0), date.today())
      with open(os.path.join(os.getcwd(), 'logs', jfnm), 'w') as jout:
        jout.write(json.dumps(mdata))
```